backend/src/services/applicationService.ts

```diff
@@ -1,40 +1,70 @@
-const apps = [
-  {
-    id: '1',
-    name: 'App1',
-    organizationId: '1',
-    updatedByUserId: '1'
-  },
-  {
-    id: '2',
-    name: 'App2',
-    organizationId: '2',
-    updatedByUserId: '2'
-  }
-];
+import { getCustomRepository } from 'typeorm';
+import { ApplicationRepository } from '../data/repositories/applicationRepository';
+import { CustomError } from '../common/models/error/CustomError';
 
-export const getApps = () => Promise.resolve(apps);
+export const getApps = async () => {
+  try {
+    const apps = await getCustomRepository(ApplicationRepository).getAllApp();
+    if (apps.length === 0) {
+      throw new CustomError('Apps not found', 404);
+    }
+    return apps;
+  } catch (e) {
+    throw new CustomError(e.msg || e.message, e.status);
+  }
+};
 
-export const getAppById = (id: string) => Promise.resolve(apps.find(a => a.id === id));
+export const getAppById = async (id: string) => {
+  try {
+    const app = await getCustomRepository(ApplicationRepository).getAppById(id);
+    if (!app) {
+      throw new CustomError('App not found', 404);
+    }
+    return app;
+  } catch (e) {
+    throw new CustomError(e.msg || e.message, e.status);
+  }
+};
 
-export const addApp = ({ appName }: any) => {
-  const appId = new Date().getTime().toString();
-  const name: string = appName;
-  const app = { id: appId, name, organizationId: `o${appId}`, updatedByUserId: `u${appId}` };
-  apps.push(app);
-  return Promise.resolve(app);
+export const addApp = async ({ name, organizationId, updatedByUserId }: any) => {
+  try {
+    const app = await getCustomRepository(ApplicationRepository).getAppByName(name);
```

Comment:
We need to check not only by name but also by user's currentOrganizationId. Two different organizations can have apps with the same name. You can get user's info from `req.user`